docs/introduction/Sidechain-SDK-extension.rst

```diff
@@ -510,4 +510,94 @@ This can be done with Guice, by binding the ""CustomApiGroups" field:
+
+The method ``createBackup`` starts the backup procedure by calling the function ``BackUpper.backup``. It takes as parameters:
+
+- ``stateStoragePath``: File path to the SidechainStateStorage.
+- ``sidechainBlockItToRollback``: Sidechain block id used for the storage rollback.
+- ``copyStateStorage``: If True performs a copy of the SidechainStateStorage before rollback it in order to avoid its permanent corruption (this is a one way procedure, the Storge can't be used anymore after the rollback).
+
+The ``BoxBackupInterface`` interface declares a method ``backup`` that should be implemented by the extending class.
+
+::
+
+    public interface BoxBackupInterface {
+        void backup(BoxIterator source, BackupStorage db) throws Exception;
+    }
+
+The ``backup`` method receives an iterator over the Storage that will be taken as a source for the backup (typically it would be the SidechainStateStorage),
+and the Storage used to store the backup.
+You can use the method ``nextBox`` from this iterator to retrieve the next non-coin box from the Storage.
+
+Important notes:
+    ::
+
+In order to maintain a consistency between what the Mainchain knows about the Sidechain and what the Sidechain contains itself, if you want to perform a Backup of a ceased Sidechain, you should rollback the Storage
+to the version that contains the Mainchain block calculated by the following formula:
+
+  Genesis_MC_block_height + (current_epch - 2) * withdrawalEpochLength - 1
+
+This can be easily done by calling the endpoint `/backup/getSidechainBlockIdForBackup <https://github.com/HorizenOfficial/Sidechains-SDK/blob/master/sdk/src/main/scala/com/horizen/api/http/SidechainBackupApiRoute.scala>`_
+and pass the block id obtained to the method ``createBackup``.
+
+Restore procedure
+-------------------------------
+
+The restore procedure is automatically invoked when a Sidechain node starts from an empty blockchain. Before the application of the genesis block, the node is able to detect if there is a Backup Storage to restore,
+and in that case, it performs a several iteration over it in order to populate the other storages.
+The Backup Storage is scanned 4 times:
+
+- First scan used to populate the ``SidechainStateStorage`` with all the boxes found in the BackupStorage.
+- Second scan used to add the boxes owned by a node wallet proposition inside the ``WalletBoxStorage``. In this way you will be able to see the restored boxes inside your wallet (only if you have the corresponding proposition imported in your wallet) and spend them.
+- Third scan performed in the application level (``ApplicationState``). This is useful if you have some custom storages that you want to populate with the information taken from these boxes. You should override the method ``public Try<ApplicationState> onBackupRestore(BoxIterator boxIterator)`` inside the ``ApplicationState``.
```

Comment:
Check `ApplicationState` and `ApplicationWallet` interfaces in the doc. We need to update them with the new methods added.

Reply:
Done